src/test/lib/storage/index/partial_hash/partial_hash_index_test.cpp

```diff
@@ -146,6 +146,36 @@ TEST_F(PartialHashIndexTest, Iterators) {
   EXPECT_NE(std::find(begin, end, RowID{ChunkID{0}, ChunkOffset{4}}), end);
 }
 
+TEST_F(PartialHashIndexTest, AccessWithIterators) {
+  auto test_access_values_with_iterators = [](auto begin, auto end) {
+    auto current_iter = begin;
+    EXPECT_EQ(current_iter, begin);
+    ++current_iter;
+    EXPECT_NE(current_iter, begin);
+    // Test size of index iterator.
+    EXPECT_EQ(std::distance(begin, end), 14);
+    // Test for not-existing value in the iterator.
+    EXPECT_NE(std::find(begin, end, RowID{ChunkID{0}, ChunkOffset{4}}), end);
+  };
+  index->access_values_with_iterators(test_access_values_with_iterators);
+
+  auto access_null_values_with_iterators = [](auto begin, auto end) {
+    // Test size of NULL values index iterator.
+    EXPECT_EQ(std::distance(begin, end), 2);
+  };
+  index->access_null_values_with_iterators(access_null_values_with_iterators);
+
+  auto access_range_equals_with_iterators = [](auto begin, auto end) {
+    EXPECT_EQ(*begin, (RowID{ChunkID{0}, ChunkOffset{0}}));
```

Comment:
We should add a test checking the distance of the range.

Reply:
Furthermore, we should add a range equals test case where more than one position is retrieved / iterated.